libledger/DBInitializer.cpp

```diff
@@ -138,6 +139,41 @@ void DBInitializer::initLevelDBStorage()
     }
 }
 
+void DBInitializer::recoverFromBinaryLog(
+    std::shared_ptr<dev::storage::BinLogHandler> _binaryLogger, dev::storage::Storage::Ptr _storage)
+{
+    int64_t num = 0;
+    auto memoryTableFactory = m_tableFactoryFactory->newTableFactory(dev::h256(), num);
+    Table::Ptr tb = memoryTableFactory->openTable(SYS_CURRENT_STATE, false);
+    if (tb)
+    {
+        auto entries = tb->select(SYS_KEY_CURRENT_NUMBER, tb->newCondition());
+        if (entries->size() > 0)
+        {
+            auto entry = entries->get(0);
+            std::string currentNumber = entry->getField(SYS_VALUE);
+            num = boost::lexical_cast<int64_t>(currentNumber.c_str());
+        }
+    }
+    auto blocksData = _binaryLogger->getMissingBlocksFromBinLog(num);
+    if (blocksData->size() > 0)
+    {
+        for (size_t i = 0; i < blocksData->size(); ++i)
+        {
+            auto blockData = blocksData->at(num + i);

```

Comment:
/// @param _currentNum : [in] block heigth in database
/// @return : missing data of each binlog block
std::shared_ptr<BlockDateMap> getMissingBlocksFromBinLog(int64_t _currentNum);

so i != 0.